### ISSUE TITLE
ObsidianExtractNote leaves text in the original buffer, new note is empty

### ISSUE THREAD

**Issue by fnune** [epwalsh/obsidian.nvim#484]
### 🐛 Describe the bug

Hi! `ObsidianExtractNote` is not working for me. With a visual selection, the text remains in the original note and the new note is empty.

Here is a video demonstrating the issue:

[Screencast from 2024-03-09 15-57-43.webm](https://github.com/epwalsh/obsidian.nvim/assets/16181067/4229c67e-a405-49e5-900c-6ceada58743e)

### Config

```
return {
  {
    "epwalsh/obsidian.nvim",
    lazy = true,
    ft = "markdown",
    dependencies = { "nvim-lua/plenary.nvim" },
    opts = {
      workspaces = {
        {
          name = "personal",
          path = "~/Downloads/test-vault",
        },
      },
      note_id_func = function(title)
        local suffix = "untitled"
        if title ~= nil then
          suffix = title:gsub(" ", "-"):gsub("[^A-Za-z0-9-]", ""):lower()
        end
        return tostring(os.date("%Y-%m-%d")) .. "-" .. suffix
      end,
      daily_notes = {
        folder = "daily",
      },
    },
    init = function()
      vim.opt.conceallevel = 2
    end,
  },
}
```

### Environment

NB: I have tried this with `version = "*"` as well as from the default branch.

```
 ~/.home.nix => nvim --version
nvim --headless -c 'lua require("obsidian").info()' -c q
NVIM v0.10.0-dev-6525832
Build type: Release
LuaJIT 2.1.1693350652
Run "nvim -V1 -v" for more info
Obsidian.nvim v3.7.0 (6888c7620109684371ed9304df2ba68abc78d558)
Status:
   buffer directory: nil
   working directory: /home/fausto/.home.nix
Workspaces:
  ✓ active workspace: Workspace(name='personal', path='/home/fausto/Downloads/test-vault', root='/home/fausto/Downloads/test-vault')
Dependencies:
   plenary.nvim: f7adfc4b3f4f91aab6caebf42b3682945fbc35be
   nvim-cmp: 04e0ca376d6abdbfc8b52180f8ea236cbfddf782
   telescope.nvim: d90956833d7c27e73c621a61f20b29fdb7122709
Integrations:
   picker: TelescopePicker()
   completion: enabled (nvim-cmp)
Tools:
   rg: ripgrep 13.0.0
Environment:
   operating system: Linux
Config:
   notes_subdir: nil%
```

**Comment by jemag** [comment]
I can reproduce

**Comment by epwalsh** [comment]
Ah, silly mistake. Fixed in dc60381.
